src/cozmo/usbmux/usbmux.py

```diff
@@ -240,64 +248,119 @@ def msg_received(self, msg):
             device_id (int): The id of the device to connect to
             port (int): The port to connect to on the target device
         Returns:
-            (asyncio.Transport, asyncio.Protocol) - The connected transport and protocol
+            (asyncio.Transport, asyncio.Protocol): The connected transport and protocol.
+                An attribute called ``device_info`` will be set on the transport
+                holding a dictionary with more information about the device,
+                such as SerialNumber.
         Raises:
             A USBMuxError subclass instance such as ConnectionRefused
         '''
         waiter = asyncio.Future(loop=self.loop)
         connector = USBMuxConnector(device_id, port, waiter)
         transport, switcher = await self._connect_transport(lambda: _ProtoSwitcher(self.loop, connector))
+        transport.device_info = self.attached.get(device_id)
 
         # wait for the connection to succeed or fail
         await waiter
 
         app_protocol = switcher.switch_protocol(protocol_factory)
         return transport, app_protocol
 
-    async def connect_to_first_device(self, protocol_factory, port, max_wait=DEFAULT_MAX_WAIT):
-        '''Open a TCP connection to the first device that has the requested port open.
+    async def wait_for_serial(self, serial, timeout=DEFAULT_MAX_WAIT):
+        '''Wait for a device with the specified serial number to attach.
 
         Args:
-            protocol_factory (callable): A callable that returns an asyncio.Protocol implementation
-            port (int): The port to connect to on the target device
-            max_wait (float): The maximum amount of time to wait for a suitable device to be connected
+            serial (string): Serial number of the device to wait for.
+            timeout (float): The maximum amount of time in secondsto wait for a
+                matching device to be connected.
+                Set to None to wait indefinitely, or -1 to only check currently
+                connected devices.
         Returns:
-            (asyncio.Transport, asyncio.Protocol) - The connected transport and protocol
+            int: The device id of the connected device
         Raises:
-            asyncio.TimeoutError - If no devices with the requested port become available in the specified time
+            asyncio.TimeoutError - If the device with the specified serial number doesn't appear.
```

Comment:
Could be adjusted to remove the dash: "asyncio.TimeoutError if the device with the specified serial number doesn't appear."